src/py42/clients/_archiveaccess/restoremanager.py

```diff
@@ -0,0 +1,171 @@
+            job_id, self._device_guid
+        )
+
+
+class RestoreJobManager(_RestorePoller):
+    def __init__(
+        self,
+        storage_archive_service,
+        device_guid,
+        archive_session_id,
+        job_polling_interval=None,
+    ):
+        super(RestoreJobManager, self).__init__(
+            storage_archive_service=storage_archive_service,
+            device_guid=device_guid,
+            job_polling_interval=job_polling_interval,
+        )
+        self._archive_session_id = archive_session_id
+
+    def get_stream(self, file_selections):
+        response = self._start_web_restore(file_selections)
+        job_id = self._wait_for_job(response)
+        return self._get_stream(job_id)
+
+    def send_stream(self, restore_path, node_guid, accepting_guid, file_selections):
+        num_files = sum([fs.num_files for fs in file_selections])
+        size = sum([fs.num_bytes for fs in file_selections])
+        return self._storage_archive_service.start_push_restore(
+            device_guid=self._device_guid,
+            accepting_device_guid=accepting_guid,
+            web_restore_session_id=self._archive_session_id,
+            node_guid=node_guid,
+            restore_groups=[
+                {u"backupSetId": -1, u"files": [f.file for f in file_selections]}
+            ],
+            restore_path=restore_path,
+            num_files=num_files,
+            num_bytes=size,
+            show_deleted=True,
+            file_permissions=u"CURRENT",
+            permit_restore_to_different_os_version=True,
+            restore_full_path=True,
+        )
+
+    def _wait_for_job(self, response):
+        job_id = response[u"jobId"]
+        while not self._is_job_complete(job_id):
+            time.sleep(self._job_polling_interval)
+        return job_id
+
+    def _is_job_complete(self, job_id):
+        response = self._storage_archive_service.get_restore_status(job_id)
+        is_done = response[u"done"]
+        percentage_dict = {
+            u"jobId": job_id,
+            u"status": response.data.get(u"status"),
+            u"percentComplete": response[u"percentComplete"] if not is_done else 100,
+        }
+        debug.logger.debug(format_dict(percentage_dict))
+        return is_done
+
+    def _start_web_restore(self, file_selections):
+        num_files = sum([fs.num_files for fs in file_selections])
+        num_dirs = sum([fs.num_dirs for fs in file_selections])
+        num_bytes = sum([fs.num_bytes for fs in file_selections])
+        return self._storage_archive_service.start_restore(
+            device_guid=self._device_guid,
+            web_restore_session_id=self._archive_session_id,
+            restore_groups=[
+                {u"backupSetId": -1, u"files": [f.file for f in file_selections]}
+            ],
+            num_files=num_files,
+            num_dirs=num_dirs,
+            num_bytes=num_bytes,
+            show_deleted=True,
```

Comment:
"show deleted" should default to False, and should also be a configurable parameter. Most people _don't_ want all their deleted files when restoring.

Reply:
`show_deleted` is and has been `True` for web restore in py42, which is why I made it True here.  @alanag13  is there a reason it's True there that you know of?

I like to keep in consistent with the default values and not have that be a breaking change, but we definitely should make it configurable.

Reply:
it's probably just true because that's what it does in the ui (or did at the time). We can make it default to false for 2.0 but changing the default would be breaking. I agree that it should be configurable though, and we could do that now if we wanted to.

Reply:
we can definitely have false be what passed in when your doing a push restore rather than a web, though.

Reply:
If it's what the browser does, then why is it unexpected for it to be default?

The only time I have ever restored is because I accidentally deleted something, not sure if that is the same
thing though

Reply:
The UI default has always been to _not_ show deleted files. And while there's definitely a use for showing deleted files, you more often than not don't want everything you've ever deleted to come back when restoring a folder or a whole system. It can get quite messy.

Reply:
You are correct, the default is supposed to be False. Also, it's false in the code: https://stash.corp.code42.com/projects/SVR/repos/core/browse/pro_core/mod_backup/src/api/com/code42/backup/webrestore/getcontents/CreateGetContentsJobCmd.java#31,37,58,77,85-86

Reply:
I also appreciate your product knowledge @timabrmsn , thanks

Reply:
I will for sure it in push restore, but if it is really kind of messy like you say, could we pass this off as a bugfix like, 

`Fix bug in web restore where we accidentally defaulted to restoring deleted files in a directory.`